motionplan/tpspace/ptgDiffDrive.go

```diff
@@ -0,0 +1,75 @@
+package tpspace
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/golang/geo/r3"
+
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// ptgDiffDrive defines a PTG family composed of circular trajectories with an alpha-dependent radius.
+type ptgDiffDrive struct {
+	maxMMPS float64 // millimeters per second velocity to target
+	maxRPS  float64 // radians per second of rotation when driving at maxMMPS and turning at max turning radius
+}
+
+// NewDiffDrivePTG creates a new PTG of type ptgDiffDrive.
+func NewDiffDrivePTG(maxMMPS, maxRPS float64) PTG {
+	return &ptgDiffDrive{
+		maxMMPS: maxMMPS,
+		maxRPS:  maxRPS,
+	}
+}
+
+// For this particular driver, turns alpha into a linear + angular velocity. Linear is just max * fwd/back.
+// Note that this will NOT work as-is for 0-radius turning. Robots capable of turning in place will need to be special-cased
+// because they will have zero linear velocity through their turns, not max.
+func (ptg *ptgDiffDrive) PTGVelocities(alpha, dist float64) (float64, float64, error) {
+	// (v,w)
+	if dist == 0 {
+		return 0, 0, nil
+	}
+	k := math.Copysign(1.0, dist)
+	v := ptg.maxMMPS * k
+	w := (alpha / math.Pi) * ptg.maxRPS * k
+	return v, w, nil
+}
+
+// Transform will return the pose for the given inputs. The first input is [-pi, pi]. This corresponds to the radius of the curve,
+// where 0 is straight ahead, pi is turning at min turning radius to the right, and a value between 0 and pi represents turning at a radius
+// of (input/pi)*minradius. A negative value denotes turning left. The second input is the distance traveled along this arc.
+func (ptg *ptgDiffDrive) Transform(inputs []referenceframe.Input) (spatialmath.Pose, error) {
+	if len(inputs) != 2 {
+		return nil, fmt.Errorf("ptgDiffDrive takes 2 inputs, but received %d", len(inputs))
+	}
+	alpha := inputs[0].Value
+	dist := inputs[1].Value
+
+	// Check for OOB within FP error
+	if math.Pi-math.Abs(alpha) > math.Pi+floatEpsilon {
+		return nil, fmt.Errorf("ptgDiffDrive input 0 is limited to [-pi, pi] but received %f", inputs[0])
```

Comment:
Can we tie this to the Limits of the frame?

Reply:
This isn't a frame and does not have Limits in the classical sense.

Reply:
The function signature is the same because it can be composed with a different data structure to expose this as the math part of a frame to do IK, but the rest of the Frame bits are not relevant here